app/src/main/java/com/example/ide/ui/viewmodel/MainViewModel.kt

```
@@ -749,12 +550,126 @@ class MainViewModel(
+                        result.onSuccess { output ->
+                            pushAssistantMessage("**Tool Output:**\n$output")
+                        }.onFailure { error ->
+                            pushAssistantMessage("Tool execution failed: ${error.message}")
+                        }
+                    }
+                    
+                    else -> {
+                        pushAssistantMessage("Action not implemented yet: $action")
+                    }
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)